addons/account/models/account_journal_dashboard.py

```diff
@@ -518,7 +518,7 @@ def open_action_with_context(self):
         ctx = dict(self.env.context, default_journal_id=self.id)
         if ctx.get('search_default_journal', False):
             ctx.update(search_default_journal_id=self.id)
-            del ctx['search_default_journal']  # otherwise it will do a useless groupby in bank statements
+            ctx['search_default_journal'] = False  # otherwise it will do a useless groupby in bank statements
```

Comment:
I don't get why this change is needed. You mention that deleting the key is not enough "in some cases"; could you be more explicit ? (also in the commit message: it's important we understand what you were doing if we come back to this piece of code with another problem in the future)

As William was saying, we're doing ctx.get('search_default_journal', False) anyway, so whether it is or not in the context dictionnary with a False value should make no difference.

Reply:
I don't have any clue.
When
* `ctx['search_default_journal'] == True` there is always the group by journal; expected
* `ctx['search_default_journal'] == False` there is never the group by journal; expected
* `ctx['search_default_journal']` is not set, from one action it works, from the other not, couldn't find why; wtf

Reply:
> We're doing ctx.get('search_default_journal', False) anyway, so whether it is or not in the context dictionnary with a False value should make no difference.

This code is already executed, the issue is for later in the web client because of the search view that is looking at the context